apiclient/harvester_api/managers.py

```diff
@@ -361,27 +362,95 @@ def delete(self, name, namespace=DEFAULT_NAMESPACE, *, raw=False):
 
 
 class BackupManager(BaseManager):
-    BACKUP_fmt = "v1/harvester/harvesterhci.io.virtualmachinebackups/{ns}/{uid}"
+    BACKUP_fmt = "v1/harvester/harvesterhci.io.virtualmachinebackups/{ns}{uid}"
     RESTORE_fmt = "v1/harvester/harvesterhci.io.virtualmachinerestores/{ns}"
 
     RestoreSpec = RestoreSpec
 
     def get(self, name="", namespace=DEFAULT_NAMESPACE, *, raw=False, **kwargs):
-        path = self.BACKUP_fmt.format(uid=name, ns=namespace)
-        return self._get(path, raw=raw, **kwargs)
+        path = self.BACKUP_fmt.format(uid=f"/{name}", ns=namespace)
+        resp = self._get(path, raw=raw, **kwargs)
+        try:
+            code, data = resp[0], resp[1]
+            if name and "backup" != data['spec']['type']:
+                return 404, dict(type='error', status=404, message=f'Backup {name!r} not found')
+
+            data['data'] = [d for d in data['data'] if "backup" == d.get('spec', {}).get('type')]
+            return code, data
+        except TypeError:
+            # raw=True
+            return resp
+        except KeyError:
+            # !data.spec || !data.data
+            return code, data
 
-    def create(self, name, restore_spec, namespace=DEFAULT_NAMESPACE, *, raw=False, **kwargs):
-        _, data = self.get(name, namespace)
-        old_vm = data['spec']['source']['name']
-        spec = restore_spec.to_dict(name, namespace, old_vm)
-        path = self.RESTORE_fmt.format(ns=restore_spec.namespace or namespace)
-        return self._create(path, json=spec, raw=raw, **kwargs)
+    def create(self, *args, **kwargs):
+        # Delegate to vm.backups
+        return self.api.vms.backup(*args, **kwargs)
+
+    def restore(self, name, restore_spec, namespace=DEFAULT_NAMESPACE, *, raw=False, **kwargs):
+        code, data = self.get(name, namespace)
+        try:
+            old_vm = data['spec']['source']['name']
+            spec = restore_spec.to_dict(name, namespace, old_vm)
+            path = self.RESTORE_fmt.format(ns=restore_spec.namespace or namespace)
+            return self._create(path, json=spec, raw=raw, **kwargs)
+        except KeyError:
+            return code, data
 
     def delete(self, name, namespace=DEFAULT_NAMESPACE, *, raw=False, **kwargs):
-        path = self.BACKUP_fmt.format(uid=name, ns=namespace)
+        path = self.BACKUP_fmt.format(uid=f"/{name}", ns=namespace)
         return self._delete(path, raw=raw, **kwargs)
 
-    restore = create  # alias
+
+class VirtualMachineSnapshotManager(BackupManager):
+    RestoreSpec = SnapshotRestoreSpec
+
+    def create_data(self, vm_uid, vm_name, snapshot_name, namespace):
```

Comment:
we use `create_data` in other managers, and consider the case:
```python
spec = api.vm_snapshots.RestoreSpec(...)  # will returns an object to manipulate

data = api.vm_snapshots.create_data(...)   # will returns a dict as a data set
spec = api.vm_snapshots.create_spec(...)  # will returns a dict instead of object
```

we already let `spec` point to an object to manipulate and propagate to API calls, so I thought in here, we use `create_data` will self-explain its type.